lib/bolero-generator/src/arbitrary.rs

```diff
@@ -80,4 +80,22 @@ mod tests {
     fn vec() {
         let _ = generator_test!(gen_arbitrary::<Vec<usize>>());
     }
+
+    #[derive(Clone, Debug, PartialEq, Eq)]
+    struct UnlikelyToBeValid(u128);
+
+    impl<'a> arbitrary::Arbitrary<'a> for UnlikelyToBeValid {
+        fn arbitrary(u: &mut arbitrary::Unstructured<'a>) -> arbitrary::Result<UnlikelyToBeValid> {
+            let v = u.arbitrary::<u128>()?;
+            if v >= 1024 {
+                return Err(arbitrary::Error::IncorrectFormat);
+            }
+            Ok(UnlikelyToBeValid(v))
+        }
+    }
+
+    #[test]
+    fn unlikely_to_be_valid() {
+        let _ = generator_test!(gen_arbitrary::<UnlikelyToBeValid>());
```

Comment:
@Ekleog I saw a failure on this test: https://github.com/camshaft/bolero/actions/runs/8257285194/job/22587562228?pr=208#step:9:522

We'll need to figure out how to make it less flaky.

Reply:
Hmm is the random number generation not uniform? 1024 / 2**128 is definitely not something that should in any way be flaky :sweat_smile:

Reply:
yeah it should be uniform... i haven't had a chance to dig into why it failed